app/utils.py

```diff
@@ -58,6 +58,28 @@ def pagination_links(pagination, endpoint, args):
     return links
 
 
+def result_meta(total_count):
+    return {"total": total_count}
+
+
+def single_result_response(result_name, result, serialize_kwargs={}):
+    return jsonify(**{result_name: result.serialize(**serialize_kwargs)})
+
+
+def list_result_response(result_name, results_query, serialize_kwargs={}):
+    meta = result_meta(results_query.count())
+    serialized_results = [result.serialize(**serialize_kwargs) for result in results_query]
+    return jsonify(meta=meta, **{result_name: serialized_results})
+
+
+def paginated_result_response(result_name, results_query, page, per_page, endpoint, request_args, serialize_kwargs={}):
+    meta = result_meta(results_query.count())
```

Comment:
I suspect this will cause the query to get executed for a *third* time. The pagination routines are bad enough in that they make a separate "count" call to calculate the page count, which is often ~90% of the effort for the db to perform as the actual query.

I would advise trying to dig in to the pagination routines to get its count out. If it makes this difficult or impossible, that probably means it needs to be hit with a hammer itself.

Reply:
Good idea. Have now implemented.